cdap-runtime-ext-dataproc/src/main/java/co/cask/cdap/runtime/spi/provisioner/dataproc/DataProcProvisioner.java

```diff
@@ -144,6 +147,22 @@ private String getMasterExternalIp(Cluster cluster) {
     return ip;
   }
 
+  @Override
+  public PollingStrategy getPollingStrategy(ProvisionerContext context, Cluster cluster) {
+    DataProcConf conf = DataProcConf.fromProperties(context.getProperties());
+    PollingStrategy strategy = PollingStrategies.fixedInterval(conf.getPollInterval(), TimeUnit.SECONDS);
+    switch (cluster.getStatus()) {
+      case CREATING:
+        strategy = PollingStrategies.initialDelay(strategy, conf.getPollCreateDelay(),
```

Comment:
just return here, instead of assigning to a variable and returning below?

Also, `CREATING` and `DELETING` are the only status that this method will get called for, right?
If so, have a `default:` case, where we `LOG.warn` saying that something is going wrong?

Reply:
I agree it's a bit confusing about what should happen for the other statuses. However, there's nothing in the API that enforces that it will only be creating or deleting. So from the provisioner's point of view, any status is valid. 

Maybe Provisioner should have one method to get the create strategy and one to get the delete strategy?

Reply:
Maybe just say that the cluster status is guaranteed to be `CREATING` or `DELETING`, if that is the case?

Reply:
yeah i guess so, will add that.